addon/templates/components/polaris-text-field.hbs

```diff
@@ -0,0 +1,118 @@
+<!-- template-lint invalid-interactive=false -->
+
+<div class="Polaris-Labelled__LabelWrapper">
+  <div class="Polaris-Label">
+    <label id="{{id}}Label" for={{id}} class="Polaris-Label__Text">{{label}}</label>
+  </div>
+  {{#if labelAction}}
+    {{polaris-button
+      plain=true
+      text=labelAction.content
+      accessibilityLabel=labelAction.accessibilityLabel
+      onClick=(action labelAction.onAction)
+    }}
+  {{/if}}
+</div>
+
+{{#polaris-text-field/connected
+  left=connectedLeft
+  right=connectedRight
+}}
+  <div
+    class={{textFieldClasses}}
+    onfocus={{action "handleFocus"}}
+    onblur={{action "handleBlur"}}
+    onclick={{action "handleClick"}}
+  >
+
+    {{#if prefix}}
+      <div class="Polaris-TextField__Prefix" id="{{id}}Prefix">
+        {{render-content prefix}}
+      </div>
+    {{/if}}
+
+    {{#if multiline}}
+      <textarea
+        name={{name}}
+        id={{id}}
+        disabled={{disabled}}
+        readonly={{readOnly}}
+        autofocus={{autoFocus}}
+        placeholder={{placeholder}}
+        onfocus={{action onFocus}}
+        onblur={{action onBlur}}
+        style={{heightStyle}}
+        autocomplete={{if autoComplete "on" "off"}}
+        class="Polaris-TextField__Input"
+        onchange={{action "handleChange"}}
+        minlength={{minLength}}
+        maxlength={{maxLength}}
+        spellcheck={{spellCheck}}
+        aria-describedby={{ariaDescribedBy}}
+        aria-labelledby={{ariaLabelledBy}}
+        aria-invalid={{ariaInvalid}}
+        value={{value}}
+      >
+      </textarea>
+    {{else}}
+      <input
+        name={{name}}
+        id={{id}}
+        disabled={{disabled}}
+        readonly={{readOnly}}
+        autofocus={{autoFocus}}
+        value={{value}}
+        placeholder={{placeholder}}
+        onfocus={{action onFocus}}
+        onblur={{action onBlur}}
+        autocomplete={{if autoComplete "on" "off"}}
+        class="Polaris-TextField__Input"
+        onchange={{action "handleChange"}}
+        min={{min}}
+        max={{max}}
+        minlength={{minLength}}
+        maxlength={{maxLength}}
+        spellcheck={{spellCheck}}
+        aria-describedby={{ariaDescribedBy}}
+        aria-labelledby={{ariaLabelledBy}}
+        aria-invalid={{ariaInvalid}}
+      >
+    {{/if}}
+
+    {{#if suffix}}
+      <div class="Polaris-TextField__Suffix" id="{{id}}Suffix">
+        {{render-content suffix}}
+      </div>
+    {{/if}}
+
+    {{#if (and (eq type "number") (not disabled))}}
```

Comment:
Where possible we try to avoid using non-standard helpers to reduce the dependency on other addons. In this case it should be easy to define a `shouldShowSpinner` CP and use that instead 👍